docs/source/data-structures/misc/libsemigroups-error.rst

```diff
@@ -0,0 +1,39 @@
+.. Copyright (c) 2024 Joseph Edwards
+
+   Distributed under the terms of the GPL license version 3.
+
+   The full license is in the file LICENSE, distributed with this software.
+
+.. currentmodule:: libsemigroups_pybind11
+
+LibsemigroupsError
+==================
+
+This page describes the custom error type used in ``libsemigroups_pybind11``,
+namely ``LibsemigroupsError``.
+
+.. autoclass:: LibsemigroupsError
+
+    All exceptions thrown by ``libsemigroups_pybind11`` are of the type
```

Comment:
I'm not sure this is completely accurate, we sometimes raise other exceptions when they arise outside the C++ code.